File: 03_square_every_digit.rb
def square_digits num
  num.digits.map {|x| x ** 2}.reverse.join.to_i
end

puts square_digits(9119)        # 911181
puts square_digits(765)         # 493625
puts square_digits(3212)        # 9414
puts square_digits(2112)        # 4114
puts square_digits(1111)        # 1111
puts square_digits(1234321)     # 14916941
puts square_digits(0)           # 0

File: 03_square_every_digit.js
function squareDigits(num) {
  return (
    String(num)
      .split("")
      .map((x) => Number(x) ** 2)
      .join("") * 1
  );
}

console.log(squareDigits(9119)); // 911181
console.log(squareDigits(765)); // 493625
console.log(squareDigits(3212)); // 9414
console.log(squareDigits(2112)); // 4114
console.log(squareDigits(1111)); // 1111
console.log(squareDigits(1234321)); // 14916941
console.log(squareDigits(0)); // 0
